session/session.go

```diff
@@ -51,6 +52,10 @@ type Session struct {
 	// Location is the probe location.
 	Location *model.LocationInfo
 
+	// PrivacySettings contains the collector privacy settings. The default
+	// is to only redact the user's IP address from results.
+	PrivacySettings collector.PrivacySettings
+
```

Comment:
Seems a good API to me: you construct session with sane defaults. You can change such defaults. In practice in probe-cli we'll always change `PrivacySettings` values. Note that probe-cli has same names and semantics (i.e. no need to apply any `!` or to map things with different names => less chance of errors.)